src/main/scala/castalia/StubService.scala

```diff
@@ -1,49 +1,63 @@
 package castalia
 
 import akka.actor.ActorSystem
+import akka.http.scaladsl.model.StatusCodes._
 import akka.http.scaladsl.server.Directives._
 import akka.http.scaladsl.server.Route
+import akka.pattern.after
 import castalia.model.ResponseConfig
 
-class StubService(theStubsByEndpoints: StubConfigsByEndpoint)(implicit val system: ActorSystem) extends Routes {
-  protected val serviceName = "StubRoutes"
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
 
-  protected def stubsByEndpoints: StubConfigsByEndpoint = theStubsByEndpoints
+class StubService(stubsByEndpoints: StubConfigsByEndpoint)(implicit val system: ActorSystem) extends Routes {
+  protected val serviceName = "StubRoutes"
 
   protected lazy val dynamicStubRoutes = {
-    def createRoute(endpoint: String, responses: ResponsesByRequest): Route = pathPrefix(endpoint) {
-      post {
-            path("responses") {
-              entity(as[ResponseConfig]) {
-                responseConfig =>
-                  responses(responseConfig.id) = (responseConfig.httpStatusCode, responseConfig.response)
-                  complete("")
+    def createRoute(endpoint: String, responses: ResponsesByRequest): Route =
+      pathPrefix(endpoint) {
+        post {
+              path("responses") {
+                entity(as[ResponseConfig]) {
+                  rc =>
+                    responses(rc.id) = (rc.delay, rc.httpStatusCode, rc.response)
+                    complete("")
+                }
               }
+        } ~
+        get {
+          path(Segment) { id =>
+            val response: Option[StubResponse] = responses.get(id)
+
+            response match {
+              case Some((Some(delay), statusCode: StatusCode, optResponse: AnyJsonObject)) =>
+                optResponse match {
+                  case Some(content) => complete(
+                    after(delay.duration, system.scheduler)
+                    (Future(statusCode, content.toJson)))
```

Comment:
The formatting of this line can be confusing. Maybe the `()` around `Future` can be replaced by `{}`